applications/incompressible_navier_stokes/navier_stokes_manufactured/application.h

```diff
@@ -0,0 +1,798 @@
+    pp_data.output_data.filename           = this->output_parameters.filename;
+    pp_data.output_data.write_divergence   = true;
+    pp_data.output_data.write_shear_rate   = true;
+    pp_data.output_data.degree             = this->param.degree_u;
+    pp_data.output_data.write_higher_order = true;
+
+    // calculation of velocity error
+    pp_data.error_data_u.write_errors_to_file               = true;
+    pp_data.error_data_u.time_control_data.is_active        = true;
+    pp_data.error_data_u.time_control_data.start_time       = start_time;
+    pp_data.error_data_u.time_control_data.trigger_interval = (end_time - start_time);
+    pp_data.error_data_u.analytical_solution.reset(new AnalyticalSolutionVelocity<dim>());
+    pp_data.error_data_u.calculate_relative_errors = true; // false;
+    pp_data.error_data_u.name                      = "velocity";
+
+    // ... pressure error
+    pp_data.error_data_p.write_errors_to_file               = true;
+    pp_data.error_data_p.time_control_data.is_active        = true;
+    pp_data.error_data_p.time_control_data.start_time       = start_time;
+    pp_data.error_data_p.time_control_data.trigger_interval = (end_time - start_time);
+    pp_data.error_data_p.analytical_solution.reset(new AnalyticalSolutionPressure<dim>());
+    pp_data.error_data_p.calculate_relative_errors = true; // false;
+    pp_data.error_data_p.name                      = "pressure";
+
+    std::shared_ptr<PostProcessorBase<dim, Number>> pp;
+    pp.reset(new PostProcessor<dim, Number>(pp_data, this->mpi_comm));
+
+    return pp;
+  }
+
+  bool                   include_convective_term                      = true;
+  bool                   pure_dirichlet_problem                       = true;
+  double                 start_time                                   = 0.0;
+  double                 end_time                                     = 0.1;
+  double                 density                                      = 1000.0;
+  double                 kinematic_viscosity                          = 5e-6;
+  bool                   use_turbulence_model                         = false;
+  bool                   use_generalized_newtonian_model              = true;
+  bool                   formulation_viscous_term_laplace_formulation = true;
+  TemporalDiscretization temporal_discretization               = TemporalDiscretization::Undefined;
+  std::string            temporal_discretization_string        = "BDFCoupledSolution";
+  bool                   treatment_of_convective_term_implicit = false;
+  bool                   treatment_of_variable_viscosity_implicit = false;
+
+  double generalized_newtonian_viscosity_margin = 49.0e-6;
+  double generalized_newtonian_kappa            = 1.1;
+  double generalized_newtonian_lambda           = 10.0;
+  double generalized_newtonian_a                = 2.1;
+  double generalized_newtonian_n                = 0.25;
```

Comment:
Why not using a struct for all these parameters? Calling/defining the constructor of the Neumann boundary condition would become much simpler/shorter.